docs/api/StreamStart.md

```diff
@@ -17,15 +17,40 @@ QUIC_STATUS
 
 # Parameters
 
-**TODO**
+`Stream`
+
+The valid handle to an open stream object.
+
+`Flags`
+
+The set of flags the controls the behavior of `StreamStart`:
+
+Value | Meaning
+--- | ---
+**QUIC_STREAM_START_FLAG_NONE**<br>0 | No special behavior. Executes as a blocking call, returning only after the operation has completed. The peer is not informed of the stream starting until the app sends data.
+**QUIC_STREAM_START_FLAG_FAIL_BLOCKED**<br>1 | Only opens the stream if flow control (from the peer) allows. If the stream count limit has currently been reached the start will fail with `QUIC_STATUS_STREAM_LIMIT_REACHED`.
+**QUIC_STREAM_START_FLAG_IMMEDIATE**<br>2 | Indicates that the peer should be immediately informed of the stream opening (or at least as soon as flow control allows) and not wait for any data to be sent first.
+**QUIC_STREAM_START_FLAG_ASYNC**<br>4 | Causes the API to be non-blocking. The API call itself will just queue the operation and immediate return. The final result of the start operation will be delivered via the `QUIC_STREAM_EVENT_START_COMPLETE` event.
+**QUIC_STREAM_START_FLAG_SHUTDOWN_ON_FAIL**<br>8 | Causes the stream to immediate shutdown (abortive) if the start operation fails.
+**QUIC_STREAM_START_FLAG_INDICATE_PEER_ACCEPT**<br>16 | Indicates the app wants the `QUIC_STREAM_EVENT_PEER_ACCEPTED` event to be delivered if the stream isn't initially accepted (allowed by flow control) at start completion.
 
 # Return Value
 
 The function returns a [QUIC_STATUS](../v0/QUIC_STATUS.md). The app may use `QUIC_FAILED` or `QUIC_SUCCEEDED` to determine if the function failed or succeeded.
 
 # Remarks
 
-**TODO**
+This function starts the processing of the stream by the connection. Once called, the stream can start receiving events to the handler passed into [StreamOpen](StreamOpen.md). If the start operation fails, the **only** event that will be delivered is `QUIC_STREAM_EVENT_START_COMPLETE` with the failure status code.
+
+The first step of the start process is assigning the stream an identifier (stream ID). The stream ID space is flow controlled, meaning the peer is able to control how many streams the app can open (on-wire). Though, even if the peer won't accept any more streams currently, this API (by default) allows the app to still start the stream and assigns a local stream ID. But in this case, the stream is just queued locally until the peer will accept it.
+
+If the app does not want the queuing behavior, and wishes to fail instead, it can use the `QUIC_STREAM_START_FLAG_FAIL_BLOCKED` flag. The `QUIC_STREAM_START_FLAG_INDICATE_PEER_ACCEPT` flag can also be used to get the `QUIC_STREAM_EVENT_PEER_ACCEPTED` event to know when the stream becomes unblocked by flow control.
```

Comment:
So using both QUIC_STREAM_START_FLAG_FAIL_BLOCKED and QUIC_STREAM_START_FLAG_INDICATE_PEER_ACCEPT is nonsensical? Is it disallowed?

Reply:
It's not explicitly disallowed, but is nonsensical. If you pass the QUIC_STREAM_START_FLAG_FAIL_BLOCKED and it's blocked, then QUIC_STREAM_START_FLAG_INDICATE_PEER_ACCEPT won't ever result in you getting the event later, because there won't be a later.

Reply:
Side issue: why not just always generate QUIC_STREAM_EVENT_PEER_ACCEPTED? I realize it's unnecessary if I always use FAIL_BLOCKED, and even if I don't I may not care. But it seems like it's cleaner to just always generate this and let the user ignore it if they don't care.

Reply:
But I suppose that's a breaking change too, so beyond the scope of this review...

Reply:
The event was recently added and in order to prevent a breaking change we had to hide it behind the new flag.